frontend/src/components/common/Resource/Resource.tsx

```diff
@@ -866,7 +866,17 @@ export function GetEnvironmentVariables(props: EnvironmentVariablesProps) {
     },
     {
       label: t('translation|From'),
-      getter: (data: any) => data.from,
+      getter: (data: any) => {
+        const [kind, name] = data.from.split('/');
+        if (!kind || !name) {
+          return data.from;
+        }
+        return (
+          <Link routeName={kind} params={{ name, namespace }}>
```

Comment:
Usually we have done linking to resources by getting the details link from the respective class. See the `makeOwnerReferences` from https://github.com/headlamp-k8s/headlamp/blob/58c51a58932408d361ee725b1af27bcfc52df7ff/frontend/src/components/common/Resource/MetadataDisplay.tsx .